src/core/analysis-context.ts

```diff
@@ -484,18 +486,19 @@ export class AnalysisContext {
   }
 
   /**
-   * Returns true if the url given is resovable by the Analyzer's `UrlResolver`.
+   * Resolves a URL with this Analyzer's `UrlResolver` or returns the given
+   * URL if it can not be resolved.
    */
-  canResolveUrl(url: PackageRelativeUrl): boolean {
-    return this.resolver.canResolve(url);
+  resolveUrl(url: PackageRelativeUrl): ResolvedUrl|undefined {
+    return this.resolver.resolve(url);
   }
 
   /**
-   * Resolves a URL with this Analyzer's `UrlResolver` or returns the given
-   * URL if it can not be resolved.
+   * Resolves all resolvable URLs in the list.
+   *
+   * Unresolvable URLs will be filtered out.
    */
-  resolveUrl(url: PackageRelativeUrl): ResolvedUrl {
-    return this.resolver.canResolve(url) ? this.resolver.resolve(url) :
-                                           (url as any as ResolvedUrl);
+  resolveUrls(urls: PackageRelativeUrl[]): ResolvedUrl[] {
+    return urls.map((u) => this.resolveUrl(u)!).filter((u) => u !== undefined);
```

Comment:
super duper nit: the `!` in `this.resolveUrl(u)!` shouldn't be necessary

Reply:
A cast after the whole expression might be more clear